### ISSUE TITLE
ALX: a function that returns the sum of all its parameters (variadic function)

### ISSUE THREAD

**Issue by ahmed149** [ahmed149/learning-programing-in-C#72]
#include "variadic_functions.h"
#include <stdarg.h>

/**
 * sum_them_all - return sum of all parameters
 * @n: number of arguments
 * Return: sum of parameters
 */
int sum_them_all(const unsigned int n, ...)
{
	va_list list;
	int sum = 0;
	unsigned int a;

	if (n == 0)
		return (0);
	va_start(list, n);
	for (a = 0; a < n; a++)
	{
		sum = sum + va_arg(list, unsigned int);
	}
	va_end(list);
	return (sum);
}

**Comment by ahmed149** [comment]
Certainly! This C function, `sum_them_all`, is a variadic function that calculates the sum of a variable number of integer arguments. It uses the `<stdarg.h>` library to work with variable-length argument lists. Here's an explanation of each part of the function:

1. `#include "variadic_functions.h"`:
   - This line includes a user-defined header file called "variadic_functions.h." It's common to include header files at the beginning of C files to declare functions and data structures used in the program.

2. `#include <stdarg.h>`:
   - This line includes the standard C library `<stdarg.h>`, which is essential for working with variadic functions.

3. `int sum_them_all(const unsigned int n, ...)`:
   - This is the function declaration for `sum_them_all`. It's defined to take a variable number of arguments represented by `...`. The `n` parameter specifies the number of expected arguments.

4. `va_list list;` and `int sum = 0;`:
   - These lines declare a `va_list` named `list`, which will be used to manage the variable-length argument list, and an integer variable `sum`, which will store the sum of the arguments.

5. `unsigned int a;`:
   - This line declares an unsigned integer variable `a`, which will be used as a loop counter.

6. `if (n == 0)`:
   - This condition checks if the number of expected arguments `n` is zero. If so, it returns 0 immediately, indicating that there are no arguments to sum.

7. `va_start(list, n);`:
   - This line initializes the `list` variable to point to the first unnamed argument after `n`. In other words, it prepares to traverse the variable-length argument list.

8. `for (a = 0; a < n; a++)`:
   - This loop iterates `n` times, where `n` is the number of expected arguments.

9. `sum = sum + va_arg(list, unsigned int);`:
   - Inside the loop, it uses `va_arg` to retrieve the next argument from the variable-length list (`list`) as an unsigned integer and adds it to the `sum`.

10. `va_end(list);`:
    - After processing all the arguments, this line cleans up the `list` variable, signaling that you've finished working with the variable-length argument list.

11. `return (sum);`:
    - The `sum_them_all` function returns the computed sum of the arguments.

In summary, this function allows you to pass a variable number of integer arguments to it, and it calculates and returns their sum. It checks for the case when `n` is zero (no arguments) and uses `va_start`, `va_arg`, and `va_end` to access the variable-length argument list safely.
